src/formatDate.js

```diff
@@ -8,7 +8,35 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const delimiters = ['-', '/', '.'];
+  const inputDelimiter = delimiters.find((d) => date.includes(d));
+  const dateParts = date.split(inputDelimiter);
+
+  const dateObj = {};
+
+  fromFormat.forEach((part, index) => {
+    dateObj[part] = dateParts[index];
+  });
+
+  if (dateObj['YYYY'] && toFormat.includes('YY')) {
+    dateObj['YY'] = dateObj['YYYY'].slice(2);
+  } else if (dateObj['YY'] && toFormat.includes('YYYY')) {
+    const year = parseInt(dateObj['YY'], 10);
+
+    dateObj['YYYY'] = year < 30 ? '20' + dateObj['YY'] : '19' + dateObj['YY'];
+  }
+
+  let outputDelimiter = '-';
+
+  if (toFormat.includes('/')) {
+    outputDelimiter = '/';
+  } else if (toFormat.includes('.')) {
+    outputDelimiter = '.';
+  }
+
+  const newDate = toFormat.map((part) => dateObj[part]).join(outputDelimiter);
+
+  return newDate.slice(0, -1);
```

Comment:
The `slice` method call at the end of this line seems unnecessary, as nothing should be appended to the `newDate` that needs to be removed. Please verify the logic here.